src/hooks/wallets/useWalletBalance.ts

```diff
@@ -0,0 +1,19 @@
+import useAsync from '../useAsync'
+import useWallet from './useWallet'
+import { useWeb3ReadOnly } from '@/hooks/wallets/web3'
+import { type BigNumber } from 'ethers'
+
+const useWalletBalance = () => {
+  const web3ReadOnly = useWeb3ReadOnly()
+  const wallet = useWallet()
+
+  return useAsync<BigNumber | undefined>(() => {
+    if (!wallet || !web3ReadOnly) {
+      return undefined
+    }
+
+    return web3ReadOnly.getBalance(wallet.address, 'latest')
```

Comment:
Unfortunately, the balance that onboard returns is already formatted as a string. We could use this in the header even though it might have a different format than our number formatting rules but @schmanu made a valid point that if we want to warn the user that their balance is not enough to execute a transaction the amount in WEI or similar raw value would be better.